src/ConnectedMode.UnitTests/Suppressions/ClientSuppressionSynchronizerTests.cs

```diff
@@ -0,0 +1,132 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System.Collections.Generic;
+using SonarLint.VisualStudio.ConnectedMode.Suppressions;
+using SonarLint.VisualStudio.Core.Suppression;
+using SonarLint.VisualStudio.Core.Suppressions;
+using SonarLint.VisualStudio.IssueVisualization;
+using SonarLint.VisualStudio.IssueVisualization.Models;
+using SonarLint.VisualStudio.TestInfrastructure;
+
+namespace SonarLint.VisualStudio.ConnectedMode.UnitTests.Suppressions
+{
+    [TestClass]
+    public class ClientSuppressionSynchronizerTests
+    {
+        [TestMethod]
+        public void MefCtor_CheckIsExported()
+        {
+            MefTestHelpers.CheckTypeCanBeImported<ClientSuppressionSynchronizer, IClientSuppressionSynchronizer>(
+                MefTestHelpers.CreateExport<IClientIssueStore>(),
+                MefTestHelpers.CreateExport<IIssuesFilter>());
+        }
+
```

Comment:
The three tests are quite similar to be honest, the reason why i made three is to first check only if an unsuppressed issue gets suppressed, the second test is for the opposite and then the third is just for checking multiple issues (However I could just remove the first 2 tests)

Reply:
Difficult to know, really - there are pros and cons for both. You definitely need the "multiple" test, and as you say it does the work for the other two.

I'd be inclined to keep them all, although you could combine the first two into a data-driven test:

```
[TestMethod]
[DataRow(true)]
[DataRow(false)]
public void SychronizeIssues_SingleIssue_IsSuppressedOnServer(bool suppressedOnServer)
{
...
}
```